src/java.base/share/classes/com/sun/crypto/provider/Poly1305.java

```diff
@@ -235,12 +243,37 @@ private void processBlock(byte[] block, int offset, int length) {
         a.setProduct(r);                // a = (a * r) % p
     }
 
+    // This is an intrinsified method. The unused parameters aLimbs and rLimbs are used by the intrinsic.
+    // They correspond to this.a and this.r respectively
+    @ForceInline
+    @IntrinsicCandidate
+    private void processMultipleBlocks(byte[] input, int offset, int length, long[] aLimbs, long[] rLimbs) {
+        while (length >= BLOCK_LENGTH) {
+            n.setValue(input, offset, BLOCK_LENGTH, (byte)0x01);
```

Comment:
You could call `processBlock(input, offset, BLOCK_LENGTH);` here.

Reply:
done (duh.. thanks, neater code)